src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java

```diff
@@ -218,10 +218,24 @@ public boolean rename(Path src, Path dst) throws IOException {
       return true;
     }
     LOG.debug("Checking if source exists {}", src);
+
     if (exists(src)) {
-      LOG.debug("Source {} exists", src);
+      boolean result = storageClient.rename(hostNameScheme, src.toString(), dst.toString());
+
+      // Check if path was a directory
+      FileStatus[] files = storageClient.list(hostNameScheme, src, true, true);
```

Comment:
@djalova Can you please call storageClient getFileStatus and see if it's directory or not? This way we will not duplicate code to figure out if the given path is dir or file. thanks